### ISSUE TITLE
Use ACE in a predictive sense?

### ISSUE THREAD

**Issue by mycarta** [partofthething/ace#14]
Fair points, thank you for elaborating on that.

I do have a couple more questions.
I went back and reviewed the literature papers; if I understand those examples correctly, I think that one could use ACE in a predictive sense in a couple of ways:
1) use the magnitude of the transforms as a measure of the strength of the relationship between the original independent predictors and the target, even though as you say, there are no functional forms for the transforms

2) predict the target given new measurements of the predictors, using the inverse relationship between theta and Y; e.g., using the example from Wang and Murphy:
![image](https://user-images.githubusercontent.com/7424763/64974831-dbf65e00-d86a-11e9-8bc6-ce07dd662ca0.png)
or in pictorial way using my example from [here](https://github.com/partofthething/ace/issues/10):
![image](https://user-images.githubusercontent.com/7424763/64973628-9173e200-d868-11e9-8540-5a7472a70b38.png)

_Originally posted by @mycarta in https://github.com/partofthething/ace/issues/11#issuecomment-531837984_

**Comment by mycarta** [comment]
hi
Do you have some insights on this?

**Comment by partofthething** [comment]
Yes that's a great use of this. That's basically how I use it. I run some simulations and build X's and Y's. Then I do continuous optimization problems choosing x's and predicting Y(x) in the way you mention. It works great for that.

**Comment by mycarta** [comment]
Do you have an example that could be added to the documentation?
I am happy to help document if needed.

**Comment by partofthething** [comment]
I may be misunderstanding, but isn't [this demo](https://partofthething.com/ace/index.html#using-it) exactly what you need for this?

**Comment by mycarta** [comment]
> I may be misunderstanding, but isn't [this demo](https://partofthething.com/ace/index.html#using-it) exactly what you need for this?

I was thinking of an example of predicting the target y given new measurement of the predictors using an existing transformation

**Comment by partofthething** [comment]
So if you could initialize the `myace = model.Model()` part by loading transformations from disk rather than re-building them with x,y pairs and then run `myace.eval([0.1, 0.2, 0.5, 0.3, 0.5])` without having to run `myace.build_model_from_xy(x, y)` then you'd be good? 

I could add `model.save()` and `model.load()` methods to help with that. Meanwhile I think just using [pickle](https://docs.python.org/3/library/pickle.html) on the model would probably work. 

Would this help?

**Comment by mycarta** [comment]
Nice example, and yes, picle would do but having a method that is part of your library would be awesome.
